packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -2359,11 +2357,15 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
    * null otherwise
    */
   private _getAriaActiveDescendantValue(): string | undefined {
-    const { selectedIndices } = this.props.hoisted;
+    const { currentOptions, selectedIndices } = this.props.hoisted;
     const { isOpen, currentPendingValueValidIndex } = this.state;
-    let descendantText = isOpen && selectedIndices?.length ? this._id + '-list' + selectedIndices[0] : undefined;
+    const options = currentOptions.map((item, index) => ({ ...item, index }));
```

Comment:
I'll make a separate PR to address this (it happens in more places within `ComboBox`).

Reply:
Actually, I get hit with an eslint rule error if I do that, so maybe we prefer this approach in v8

![image](https://user-images.githubusercontent.com/7798177/205179765-22c1f67f-8bca-4222-9c67-86c683aa9935.png)

Reply:
[The docs for this plugin](https://eslint.org/docs/latest/rules/prefer-object-spread) state that

> Introduced in ES2018, object spread is a declarative alternative which may perform better than the more dynamic, imperative Object.assign.

which, along with readability, seem to be the arguments in favor of enforcing spread. In the benchmark I linked spread is ~80% slower in Edge and ~85% slower in Chrome, while `Object.assign` is ~6% slower in Firefox. _May perform better_ indeed 😉. What I'm getting from this is that the performance is implementation-dependent so it could change tomorrow, but that even in the case where spread is faster it's not that much faster (and it's within the margin of error, +/- ~7%). Compared with the cases where spread is slower, it's _much_ slower and well outside the margin of error. So where we think performance is critical we should prefer Object.assign.

So is performance critical here? That's not so clear to me. I'm of the opinion that we should strive for Fluent to be as fast as (reasonably) possible, leaving as many resources to our users as possible so in that sense all performance is always critical. Yet, it's not clear to me that this measurably affects the perf of `ComboBox`?

I think this change is simple enough that it's worth benchmarking spread vs object.assign in `ComboBox` itself to see what shakes out.

What do you think about running a `ComboBox` bench for this case?